packages/src/context/provider.tsx

```diff
@@ -53,14 +55,27 @@ type ContentOverlayControllerProps = {
 
 function ContentOverlayController({
   isOpen,
+  current,
   overlayId,
   onMounted,
   onCloseModal,
   onExitModal,
   controller: Controller,
 }: ContentOverlayControllerProps) {
+  const prevCurrent = useRef(current);
   const onMountedRef = useRef(onMounted);
 
+  /**
+   * @description Executes when closing and reopening an overlay without unmounting.
+   */
+  if (prevCurrent.current !== current) {
+    prevCurrent.current = current;
+
+    if (current === overlayId) {
+      onMountedRef.current();
+    }
+  }
```

Comment:
This part needed some work too. I learned it thanks to you. 🧐

Reply:
I don't understand why this logic is needed.
Could anyone explain the reason? Are there the edge cases?

Reply:
The reason this is necessary is because the handling of open modals is performed during the ADD stage in the reducer.

Let’s examine the flow when an overlay is first opened:

1. Calling overlay.open triggers `dispatch(overlays, { type: 'ADD' })`.
2. The reducer then adds the `overlayId` to the `overlayOrderList`.
3. The `overlayId` added to the `overlayOrderList` renders the `ContentOverlayController` component.
4. The `ContentOverlayController` component, via `useEffect`, calls `dispatch(overlays, { type: 'OPEN' })` for the newly added overlay.

However, if the overlay is closed without unmounting, the `overlayId` is not re-added to the `overlayOrderList`, and the key remains the same. Therefore, it is treated as the same component as before, and step 4 does not occur, resulting in the `isOpen` value not changing to `true`.

To address the edge case where an overlay is closed without unmounting and then reopened, the `ContentOverlayController` now checks the `current` value, which indicates the topmost open overlay.
If the current value matches the overlayId that the `ContentOverlayController` holds, it recognizes that the `ADD` reducer has been executed and proceeds to trigger the `OPEN` reducer.